modules/contact/src/MechanicalContactConstraint.C

```diff
@@ -137,6 +141,9 @@ MechanicalContactConstraint::MechanicalContactConstraint(const InputParameters &
     _master_slave_jacobian(getParam<bool>("master_slave_jacobian")),
     _connected_slave_nodes_jacobian(getParam<bool>("connected_slave_nodes_jacobian")),
     _non_displacement_vars_jacobian(getParam<bool>("non_displacement_variables_jacobian")),
+    _contact_linesearch(dynamic_cast<ContactLineSearch *>(_fe_problem.customLineSearch())),
+    _current_contact_state(_contact_linesearch ? _contact_linesearch->contact_state()
+                                               : std::make_shared<std::set<dof_id_type>>()),
```

Comment:
I realize this is difficult wedging a new system in. It looks like you are asking the nonlinear problem if we are using one of the new custom built LineSearch objects here and making some decisions about how to handle the case where we are not. Perhaps a better design will be to build a default LineSearch object always now that we are adding this new system so that extensions to the system don't have to write logic like this? It's a little bit weird since you are messing with creating dynamic memory and setting iteration counts, etc.